src/openapi/streaming/subscription.ts

```diff
@@ -290,15 +287,28 @@ class Subscription {
                     servicePath: this.servicePath,
                     isServerInitiated,
                 });
-                this.setState(this.STATE_PUBLISHER_DOWN);
+                const referenceId = this.referenceId;
 
                 this.waitForPublisherToRespondTimer = window.setTimeout(() => {
                     this.waitForPublisherToRespondTimer = null;
-                    this.setState(this.STATE_UNSUBSCRIBED);
-                    this.tryPerformAction(ACTION_SUBSCRIBE);
+
+                    // only if nothing has changed - subscribed or mid-subscribed and the same reference id we got all the resets on
+                    if (
+                        referenceId === this.referenceId &&
+                        (this.currentState === this.STATE_SUBSCRIBED ||

```

Comment:
Suggestion:  maybe we could use `Array.includes` to check for currentState, like
```[this.STATE_SUBSCRIBED, this.STATE_SUBSCRIBE_REQUESTED, this.STATE_PATCH_REQUESTED, this.STATE_REPLACE_REQUESTED].includes(this.currentState)```

Reply:
thanks, it is a little bit more readable.